qualang_tools/plot/Fitting/fitting.py

```diff
@@ -31,154 +32,321 @@ def linear(x, y):
+                    np.exp(-x / (guess_T2 * a1))
+                    * (a5 * initial_offset / peaks[0] * 2 + np.cos(2 * np.pi * a0 * guess_freq * x + a3))
+            )
+
+        popt, pcov = scipy.optimize.curve_fit(
+            func,
             x,
             y,
-            [1, 1, 1, angle0, 1, 1, 1],
+            p0=[1, 1, 1, guess_phase, 1, 1],
         )
 
-        print(
-            f"f = {popt[0] * omega / (2 * np.pi) * xn[1]}, phase = {popt[3] % (2 * np.pi) * xn[1]}, T2 = {(1 / (decay_guess * popt[1])) * xn[1]}, amp = {peaks[0] * popt[2] * yn[1]}, uncertainty population = {post_decay_mean * popt[4] * yn[1]}, initial offset = {popt[5] * initial_offset * yn[1]}"
-        )
+        perr = np.sqrt(np.diag(pcov))
+
         out = {
-            "fit_func": lambda x: fit_type(x, popt) * yn[1],
-            "f": popt[0] * omega / (2 * np.pi) * xn[1],
-            "phase": popt[3] % (2 * np.pi) * xn[1],
-            "T2": (1 / (decay_guess * popt[1])) * xn[1],
-            "amp": peaks[0] * popt[2] * yn[1],
-            "uncertainty_population": post_decay_mean * popt[4] * yn[1],
-            "initial_offset": popt[5] * initial_offset * yn[1],
+            "fit_func": lambda x: fit_type(x / x_normal, popt) * y_normal,
+            "f": [popt[0] * guess_freq / x_normal, perr[0] * guess_freq / x_normal],
+            "phase": [popt[3] % (2 * np.pi), perr[3] % (2 * np.pi)],
+            "T2": [(guess_T2 * popt[1]) * x_normal, perr[1] * guess_T2 * x_normal],
+            "amp": [peaks[0] * popt[2] * y_normal, perr[2] * peaks[0] * y_normal],
+            "initial_offset": [popt[5] * initial_offset * y_normal, perr[5] * initial_offset * y_normal],
+            "final_offset": [post_decay_mean * popt[4] * y_normal, perr[4] * post_decay_mean * y_normal]
         }
-
+        if verbose:
+            print(
+                f"Fit results:\n"
+                f"f = {out['f'][0] * 1000:.3f} +/- {out['f'][1]:.3f} MHz, \n"
+                f"phase = {out['phase'][0]:.3f} +/- {out['phase'][1]:.3f} rad, \n"
+                f"T2 = {out['T2'][0]:.2f} +/- {out['T2'][1]:.3f} ns, \n"
+                f"amp = {out['amp'][0]:.2f} +/- {out['amp'][1]:.3f} a.u., \n"
+                f"initial offset = {out['initial_offset'][0]:.2f} +/- {out['initial_offset'][1]:.3f}, \n"
+                f"final_offset = {out['final_offset'][0]:.2f} +/- {out['final_offset'][1]:.3f} a.u."
+
+            )
+        if plot:
+            plt.plot(x_data, fit_type(x, popt) * y_normal)
+            plt.plot(x_data, y_data, ".", label=f"T2  = {out['T2'][0]:.1f} +/- {out['T2'][1]:.1f}ns")
+            plt.xlabel("Waiting time [ns]")
+            plt.ylabel("I & Q amplitude [a.u.]")
+            plt.legend(loc="upper right")
+
+        if save:
+            fit_params = dict(itertools.islice(out.items(), 1, len(out)))
+            for key in fit_params:
+                fit_params[key] = fit_params[key]
```

Comment:
Why do you need this? I don't see what it does

Reply:
You're right .. I will remove this